src/overlay/Peer.cpp

```diff
@@ -49,6 +51,38 @@ static std::string const AUTH_ACTION_QUEUE = "AUTH";
 using namespace std;
 using namespace soci;
 
+namespace
+{
+// Check the signature(s) in `tx`, adding the result to the signature cache in
+// the process. For performance reasons, this function should only be called
+// from the overlay thread.
+void
+checkTransactionSignature(AppConnector& app, TransactionFrameBaseConstPtr tx)
+{
+    ZoneScoped;
+    releaseAssert(app.getConfig().EXPERIMENTAL_BACKGROUND_TX_SIG_VERIFICATION &&
+                  app.threadIsType(Application::ThreadType::OVERLAY));
+
+    auto const& hash = tx->getContentsHash();
+    auto const& signatures = txbridge::getSignatures(tx->getEnvelope());
+    auto const snapshot = app.getOverlayThreadSnapshot();
+    BucketSnapshotState bucketSnapshot(snapshot);
```

Comment:
For consistency, please use a wrapper for read-only ledger snapshots `LedgerSnapshot`

Reply:
Done!